sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlSerializer2.scala

```diff
@@ -190,12 +193,15 @@ private[sql] object SparkSqlSerializer2 {
         case array: ArrayType => return false
         case map: MapType => return false
         case struct: StructType => return false
-        case _ =>
+        case NullType => // Do nothing
+        case _ => if (allNulTypes) allNulTypes = false // This is not a NullType.
```

Comment:
For completeness's sake, don't we also need to set this in the struct type etc cases above?